### ISSUE TITLE
@property in recursive generic protocol

### ISSUE THREAD

**Issue by rasmuse** [microsoft/pyright#1330]
This issue is a continuation of #1326.

**Describe the bug**
Like in #1326 I am creating a generic protocol with a method that returns something related to the type of `self`. The example from #1326 is now solved in pyright 1.1.98,  but if we change the method to a `@property` it does not work.

Below, I have slimmed down to a more minimal example than last time.

**To Reproduce**
```python
from dataclasses import dataclass
from typing import Protocol, TypeVar

T = TypeVar("T")


class Proto(Protocol):
    @property
    def f(self: T) -> T:
        ...


@dataclass
class Concrete:
    @property
    def f(self: "Concrete") -> "Concrete":
        return self


x: Proto = Concrete()
```

**Expected behavior**
It should type check fine.

**Actual behavior**

Pyright 1.1.98 as well as 1.1.97 (command line) says:

```
  20:12 - error: Expression of type "Concrete" cannot be assigned to declared type "Proto"
    "f" is an incompatible type
      Type "Concrete" cannot be assigned to type "T@f" (reportGeneralTypeIssues)
1 error, 0 warnings, 0 infos
```

**Additional context**

Remove the `@property` so that `f` is just a method, then
- pyright 1.1.98 passes
- pyright 1.1.97 says (quite similar to #1326 ):
```
  20:12 - error: Expression of type "Concrete" cannot be assigned to declared type "Proto"
    "f" is an incompatible type
      Type "(self: Concrete) -> Concrete" cannot be assigned to type "(self: T@f) -> T@f"
        Parameter 2: type "T@f" cannot be assigned to type "Concrete"
          "object" is incompatible with "Concrete"
        Function return type "Concrete" is not compatible with type "T@f"
          Type "Concrete" cannot be assigned to type "T@f" (reportGeneralTypeIssues)
1 error, 0 warnings, 0 infos
```

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is now addressed in Pyright 1.1.99, which I just published. It will also be included in the next release of Pylance.
